Makefile

```diff
@@ -160,3 +134,83 @@ golangci-lint-fix.%: $(GOLANGCI_LINT)
 
 .PHONY: FORCE # use this to force phony behavior for targets with pattern rules
 FORCE:
+
+##@ Deployment
+
+.PHONY: cluster
+cluster: kind ctlptl ## Create Kind cluster and local registry
+	$(CTLPTL) apply -f ctlptl.yaml
+
+.PHONY: cluster-reset
+cluster-reset: kind ctlptl ## Delete Kind cluster
+	$(CTLPTL) delete -f ctlptl.yaml
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
+.PHONY: deploy
+deploy: .gen kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
```

Comment:
One of the effects of `.gen` is codegen which never caches and takes several seconds to run, unless I'm mistaken. My opinion here is that it should be reasonable for us to assume that users of `deploy` should have already run `make generate` or whatever generation targets they need before deploying. 

When developing on Rook, we use that assumption, and I like that it allows me to iterate on deployments quickly after I have finished API changes, or when no API changes have occurred.

Reply:
Agreed, I’ll fix that.